### PR TITLE
added TPU support

### PR DESCRIPTION
**Pull request by elyesmanai**
this now works on TPU. To use TPU, the user needs to execute the following commands first:
`
export TPU_IP_ADDRESS="your-tpu-ip-address" && export XRT_TPU_CONFIG="tpu_worker;0;$TPU_IP_ADDRESS:8470"
`

then pass --use_tpu True in the command line argument when running the train.py file (or add it manually to train_args)

### CODE REVIEW

**Comment by AliOsm** [comment]
I tried this change on Google Colab TPU/GPU with latest package version and finetuned `bert-base-cased` model on wikitext-2 (36,718 examples):

GPU:
Type: Tesla P100 (Best GPU available on Google Colab).
Batch size: 64 (Maximum batch size based on memory size).
Perplexity: 3.9824.
Evaluation loss: 1.3818760017553966.
Training time: 4:40 minutes.

TPU:
Type: TPUv2.
Batch size: 64 (Maximum batch size based on memory size).
Perplexity: 4.0293.
Evaluation loss: 1.3935987635765201.
Training time: 5:23.

This PR could be great improvement to the package, the PR should be optimized more (e.g use all TPU cores) but it will make huge improvements for sure.

**Comment by ThilinaRajapakse** [comment]
I thought this wasn't working properly. @elyesmanai can you update, please?

**Comment by AliOsm** [comment]
My Colab notebook: https://colab.research.google.com/drive/1qOMo4eevugndXn-o9ndD37Fok8XGnDUN

**Comment by AliOsm** [comment]
Hello everyone, any updates here? I'm planning to use this feature to train GPT-2 from scratch, but I need a confirmation that this is the right way to implement TPU support.

**Comment by elyesmanai** [comment]
hello there, sorry for the late reply. The support I added is especially made for Google Cloud TPUs and not Google Colab.
